src/components/StateDetails.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable react/jsx-one-expression-per-line */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchAirQualityAsync } from '../redux/features/countrySlice';
+
+const StateDetails = () => {
+  const { selectedState, airQualityData } = useSelector(
+    (state) => state.country,
+  );
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (selectedState) {
+      dispatch(fetchAirQualityAsync(selectedState));
+    }
+  }, [selectedState, dispatch]);
+
+  return (
+    <section className="state-air-info">
+      <div className="state-air-details">
+        <h1>State Details</h1>
+        <p>
+          Name:
+          {selectedState?.stateName}
+        </p>
+      </div>
+
+      {airQualityData ? (
+        <div>
+          <div className="air-quality-index">
+            <h2>Air Quality Data</h2>
+            <p>Air Quality Index (AQI): {airQualityData.list[0].main.aqi}</p>
+          </div>
+
+          <h3>Components in μg/m3:</h3>
+          <ul className="air-components">
+            <li className="air-components-stat">
+              <p> Carbon monoxide :</p>
+              <p>{airQualityData.list[0].components.co}</p>
+            </li>
+            <li className="air-components-stat">
+              <p>Nitrogen monoxide:</p>
+              <p>{airQualityData.list[0].components.no}</p>
+            </li>
+            <li className="air-components-stat">
+              <p>Nitrogen dioxide:</p>
+              <p>{airQualityData.list[0].components.no2}</p>
+            </li>
+            <li className="air-components-stat">
+              <p> Ozone:</p>
+              <p>{airQualityData.list[0].components.o3}</p>
```

Comment:
- [ ] The details page design doesn't match the given design. Kindly follow the design given and make changes to the elements.